crates/outline_panel/src/outline_panel.rs

```diff
@@ -4718,6 +4919,45 @@ impl OutlinePanel {
             })
             .collect()
     }
+
+    fn has_outline_children(&self, outline: &OutlineEntryOutline, _cx: &App) -> bool {
+        // Check the actual excerpt outlines to see if this outline has children
+        if let Some(excerpts) = self.excerpts.get(&outline.buffer_id) {
+            if let Some(excerpt) = excerpts.get(&outline.excerpt_id) {
+                let outlines: Vec<_> = excerpt.iter_outlines().collect();
+
+                // Find the current outline
+                for (i, o) in outlines.iter().enumerate() {
+                    if o.range == outline.outline.range && o.depth == outline.outline.depth {
+                        // Check if the next outline has greater depth
+                        if let Some(next_outline) = outlines.get(i + 1) {
+                            if next_outline.depth > outline.outline.depth {
+                                return true;
+                            }
+                        }
+                        break;
+                    }
+                }
+            }
+        }
+        false
+    }
+
+    fn is_expandable(&self, entry: &PanelEntry, cx: &App) -> bool {
```

Comment:
```suggestion
    fn can_toggle(&self, entry: &PanelEntry, cx: &App) -> bool {
```

or something, as we do not really expand, but also collapse.